spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/RoundRobinLoadBalancer.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2013-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.loadbalancer.core;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import reactor.core.publisher.Mono;
+
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.loadbalancer.reactive.DefaultResponse;
+import org.springframework.cloud.client.loadbalancer.reactive.EmptyResponse;
+import org.springframework.cloud.client.loadbalancer.reactive.Request;
+import org.springframework.cloud.client.loadbalancer.reactive.Response;
+import org.springframework.cloud.client.loadbalancer.reactive.ServiceInstanceSupplier;
+import org.springframework.cloud.loadbalancer.support.LoadBalancerClientFactory;
+import org.springframework.core.env.Environment;
+
+/**
+ * @author Spencer Gibb
+ */
+public class RoundRobinLoadBalancer implements ReactorLoadBalancer<ServiceInstance> {
+
+	private static final Log log = LogFactory.getLog(RoundRobinLoadBalancer.class);
+
+	private final AtomicInteger nextServerCyclicCounter = new AtomicInteger(-1);
```

Comment:
Depending on whether the list of servers is ordered or not, it would be useful to init the counter with a random initial seed instead of the same seed for all clients, to avoid synchronized requests to backend servers.

Reply:
Sure, I literally grabbed this from https://github.com/Netflix/ribbon/blob/master/ribbon-loadbalancer/src/main/java/com/netflix/loadbalancer/RoundRobinRule.java

How would you get the seed without knowing a size?

Reply:
Some reasonably bounded random number should do, the modulo later will take care of it. Here's an [example](https://github.com/Netflix/ocelli/blob/master/ocelli-core/src/main/java/netflix/ocelli/loadbalancer/RoundRobinLoadBalancer.java) of an attempted rewrite for netflix loadbalancers. It is abandoned later but some of the concepts are still interesting.